### PR TITLE
E2E test: New module to test the integration with Suricata

### LINKED ISSUE #3079: E2E design and implementation: New test to check the AWS infrastructure monitoring
|Version|Component| QA development branch |
|-|-|-|
| v4.4 | Manager | [3079-e2e-test-aws-monitoring](https://github.com/wazuh/wazuh-qa/tree/3079-e2e-test-aws-monitoring) |

## Introduction

The module developed in this issue aims to test the [AWS infrastructure monitoring](https://documentation.wazuh.com/current/proof-of-concept-guide/aws-infrastructure-monitoring.html). The [research](https://github.com/wazuh/wazuh-qa/issues/2835) was made previously to define the requirements for this development.

## Development stage

- [x] In research
- [x] Planning and estimating
- [x] In development
- [x] In testing phase 
- [x] In Review
- [x] Completed

## Use cases

WIP.

### LINKED ISSUE #3114: E2E design and implementation: Suricata
## Description

The test research was accomplished in [this issue](https://github.com/wazuh/wazuh-qa/issues/2877). So, in this issue, we will design and develop the E2E test to cover those use cases.

## Tasks

- [x] Design the test
- [x] Implement the test locally
- [ ] Refactor the test for it to work in the test environment

## Conclusion

The test was implemented in [3114-e2e-test-suricata](https://github.com/wazuh/wazuh-qa/compare/3114-e2e-test-suricata). Only 1 rule of Suricata was tested, this one is inside of the `emerging-policy.rules` file.

### PR DESCRIPTION
**Pull request by mauromalara**
|Related issue|
|-------------|
|  Close #3114 |

## Description

The `test_suricata_integration.py` module checks if alerts are triggered in the manager when an event from Suricata logs matches a rule. Only 1 rule of Suricata was tested, this one is inside of the `emerging-policy.rules` file.

## Use cases

<details><summary>Suricata</summary>

**Preconditions**:
- Create an inventory as follows:
  ```
  all:
  hosts:
    wazuh-manager:
      ansible_connection: ssh
      ansible_user: USER
      ansible_password: PASSWORD
      ansible_ssh_private_key_file: PATH_TO_PRIVATE_KEY
      ansible_python_interpreter: /usr/bin/python
      dashboard_user: WAZUH_DASHBOARD_USER
      dashboard_password: WAZUH_DASHBOARD_PASS
  ```

**How to run**: `python -m pytest -s tests/end_to_end/test_suricata_integration/ --inventory_path <PATH_TO_INVENTORY>`

</details>

### Added

- `test_suricata_integration.py` this module checks if alerts are triggered in the manager when an event from Suricata logs matches a rule.

---

## Testing performed [Code analysis](https://github.com/wazuh/wazuh-qa/actions/workflows/code_analysis.yaml)

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @mauromalara (Developer)  |  end_to_end/test_suricata_integration | 🚫 🚫 🚫 | [🟢](https://github.com/wazuh/wazuh-qa/files/9269111/r1-CentOS-manager-3114.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9269112/r2-CentOS-manager-3114.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/9269113/r3-CentOS-manager-3114.zip) | Ubuntu | fa5a578 | Nothing to highlight |
| @juliamagan (Reviewer)   |    test_suricata_integration/       | 🚫 🚫 🚫 | [🟢 ](https://github.com/wazuh/wazuh-qa/files/9280041/R1-3130-test-suricata-integration.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9280042/R2-3130-test-suricata-integration.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/9280043/R3-3130-test-suricata-integration.zip)  | Ubuntu | a552039 Nothing to highlight |

### CODE REVIEW

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: tests/end_to_end/test_suricata_integration/data/test_cases/cases_suricata_integration.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- name: Generate a specific web request known to trip NIDS rules
```
[the remark this answers is not in the document]
Comment (mauromalara) [reply tests/end_to_end/test_suricata_integration/data/test_cases/cases_suricata_integration.yaml]: Done in 9243172

**Comment by mauromalara** [comment]
## Update 02/08/2022
Now the test should be run using Ubuntu on the node managed by Ansible (wazuh-manager).

File: tests/end_to_end/test_suricata_integration/data/playbooks/configuration.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      replace:
+        path: "{{ suricata_conf_path }}"
+        regexp: 'default-rule-path: \S.*\n\nrule-files:\n.*- suricata.rules'
+        replace: 'default-rule-path: /etc/suricata/rules\n\nrule-files:\n  - "*.rules"'
+
+    - name: Start Suricata
+      systemd:
+        daemon_reload: true
+        enabled: true
+        state: started
```
Comment (juliamagan) [inline tests/end_to_end/test_suricata_integration/data/playbooks/configuration.yaml:71]: We need to restart Suricata

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: tests/end_to_end/test_suricata_integration/data/playbooks/configuration.yaml (hunk 2 above)
Comment (mauromalara) [reply tests/end_to_end/test_suricata_integration/data/playbooks/configuration.yaml:76]: Done in aff0d49, the task has not changed, but the way Suricata is installed has.

**Comment by mauromalara** [comment]
## Update 04/08/2022

I have executed the test multiple times using the Ubuntu AMI available in AWS, and it passes successfully.

**Review by juliamagan — APPROVED** [review]

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**